amp/TEMPLATE/reampv2/src/modules/geocoder/components/layout/panel/GeocoderPanel.jsx

```diff
@@ -30,14 +20,15 @@ class GeocoderPanel extends Component {
     };
 
     onSelectActivity = (isSelected, activityId) => {
-        let selectedActivities = Array.from(this.state.selectedActivities);
-        if (isSelected) {
-            selectedActivities.push(activityId);
-        } else {
-            selectedActivities = selectedActivities.filter(id => id !== activityId);
-        }
-
-        this.setState({selectedActivities});
+        this.setState(previousState => {
+            let selectedActivities = Array.from(this.state.selectedActivities);
```

Comment:
instead of this.state... it should be previousState that's how you guarantee that it takes the last available state